src/ui/components/form/InputFile.tsx

```diff
@@ -61,10 +63,40 @@ export function InputFile({
   }, [onRemove, propsFile]);
 
   useEffect((): void => {
+    const params = new URL(window.location.href).searchParams;
+    const patronCodeHash = params.get('patron');
+
+    if (patronCodeHash && file?.name !== 'patron-contract.json') {
+      const metadataPromise = getPatronMetadata('metadata', patronCodeHash);
+      const wasmPromise = getPatronMetadata('wasm', patronCodeHash);
```

Comment:
How about just waiting for both at the same time with `Promise.all([metadataPromise,wasmPromise]).then()`

If both are always fetched anyways and constructs this file, why not move it all into the `getPatronMetadata` function? This would leave this hook way cleaner.

It's a nice service and cool feature to be able to one click deploy a certain contract. However, I'm not yet a big fan of this being in the `InputFile` component directly. I see ux ease of having it here, however it's some magic happening here, using a third party resource, without on a certain without any feedback to the user.
A plain button "Add Metadata from Patron" comes to mind but yes that's not as streamlined. 🤔

Reply:
> This would leave this hook way cleaner.
Yes, I refactored it =)

> this being in the InputFile component
It works as follows: in our tool, we click on the button, and then it opens contracts-ui page with fetched metadata. So, this functionality is needed only for our tool. We won't change anything in traditional contracts-ui user flow. We don't need a button or something. And I put it in the `InputFile` component because of `useEffect` and `onDrop`: I can fetch metadata before page is loaded only if the link contains "patron" parameter.